pkg/scheduler/scheduler.go

```diff
@@ -560,6 +560,7 @@ func (sched *Scheduler) scheduleOne() {
 		if !permitStatus.IsSuccess() {
 			var reason string
 			if permitStatus.Code() == framework.Unschedulable {
+				metrics.PodScheduleFailures.Inc()
```

Comment:
This doesn't look correct. For each scheduling cycle, we should at most log one failure - which is conditionally performed at:

https://github.com/kubernetes/kubernetes/blob/19e85a9092399b065309aa5b4aec71a5b8101bab/pkg/scheduler/scheduler.go#L504

We don't and shouldn't log another failure here.

Reply:
Thank you for your review.

> For each scheduling cycle, we should at most log one failure - which is conditionally performed at: https://github.com/kubernetes/kubernetes/blob/19e85a9092399b065309aa5b4aec71a5b8101bab/pkg/scheduler/scheduler.go#L504

But that path immediately returns, so each scheduling cycle logs at most once. According to the definition of metrics, I think we should increment `scheduleAttempts` with any label in one scheduling cycle.
https://github.com/kubernetes/kubernetes/blob/22cf3ca0a536b402a00ca771960702c811a3ca5c/pkg/scheduler/metrics/metrics.go#L58-L63

In this case, I think `unschedulable` is appropriate.

Reply:
Hmm... I'm a little surprised we allow/expect Permit plugin to return "Unschedulable" b/c Permit/Bind plugins belong to "Binding Cycle" which shouldn't be considered as a "schedule failure", in my opinion.

But based on current codebase, your PR is correct, we should increase PodScheduleFailures metric by 1 in the `if` block. Thanks.

I will follow up with @ahg-g to see if we should eliminate the logic related with tolerating/expecting Premit/Bind plugins to return "unschedulable" status.

Reply:
Yeah, Permit unfortunately is different. When a pod is rejected (by other pod, or timeout) this is not considered a scheduling Error, and so we used unschedulable. Do you have other suggestions we might be able to handle this case?